subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/http/HttpAuthenticationDependencyResolutionIntegrationTest.groovy

```diff
@@ -162,6 +164,47 @@ task listJars {
         'basic and digest' | 'authentication { basic(BasicAuthentication)\ndigest(DigestAuthentication) }' | DIGEST            | ['Basic', 'Digest']
     }
 
+    @Unroll
+    @Issue("gradle/gradle#5571")
+    public void "can resolve dependencies from HTTP Maven repository authenticating with HTTP header"() {
```

Comment:
⭕️ Looking at this and `TestHttpHeaderAuthenticator`, it seems that we are not actually verifying that authentication would work. Or am I missing something?

Reply:
@oehme, that's true, there is no authentication test here, but the tests makes sure that the expected headers are sent with the HTTP request. And that's what this feature wants to achieve.

Knowing that my repository works if I send a certain header, I want gradle to send this certain header.

And as there is no such functionality neither in Apache's http library nor in org.mortbay.jetty (where the *Authenticator classes come from), I created this `TestHttpAuthenticator`, although this class is not doing anything, and it does not need to, as the headers are checked elsewhere.

Is this ok?

Reply:
Yes, I was just making sure there is no better way to test this using our embedded server.